From ad0c4472bfc2492cd6cc6fc718577eac8eac4258 Mon Sep 17 00:00:00 2001
From: Petri Hintukainen <phintuka@gmail.com>
Date: Thu, 6 Aug 2026 17:32:05 +0300
Subject: [PATCH] Use correct angle.

title->angle may be different from clip->nagle (unchecked or too large).
---
 src/libbluray/bdnav/navigation.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libbluray/bdnav/navigation.c b/src/libbluray/bdnav/navigation.c
index b3f719f4..fe644544 100644
--- a/src/libbluray/bdnav/navigation.c
+++ b/src/libbluray/bdnav/navigation.c
@@ -549,7 +549,7 @@ _fill_mark(const NAV_TITLE *title, NAV_MARK *mark, int entry)
     clip = &title->clip_list.clip[mark->clip_ref];
     if (clip->cl != NULL && mark->clip_ref < title->pl->list_count) {
         mark->clip_pkt = clpi_lookup_spn(clip->cl, plm->time, 1,
-            title->pl->play_item[mark->clip_ref].clip[title->angle].stc_id);
+            title->pl->play_item[mark->clip_ref].clip[clip->angle].stc_id);
     } else {
         mark->clip_pkt = clip->start_pkt;
     }
